python/src/scipp/compat/mantid.py

```diff
@@ -688,17 +700,17 @@ def from_mantid(workspace, **kwargs):
             workspace, monitor_ws = mantid.ExtractMonitors(workspace)
             workspaces_to_delete.append(workspace)
             workspaces_to_delete.append(monitor_ws)
-        scipp_obj = convert_Workspace2D_to_data_array(workspace, **kwargs)
-    elif workspace.id() == 'EventWorkspace':
+        scipp_obj = convert_Workspace2D_to_data_array(
+            workspace, as_mask_workspace=is_mask_ws, **kwargs)
```

Comment:
Wouldn't it be nicer to check the ID in `convert_Workspace2D_to_data_array` so all the flag handling can be removed?